### PR TITLE
Font Library: make font collection fields translatable

### PR DESCRIPTION
**Pull request by matiasbenedetto**
## What?
Porting changes back from WordPress core PR already merged: https://github.com/WordPress/wordpress-develop/pull/6130

### CODE REVIEW

**Review by vcanales — APPROVED** [review]
LGTM

File: lib/compat/wordpress-6.5/fonts/class-wp-font-collection.php [hunk 1]
```diff
@@ -48,28 +48,32 @@ final class WP_Font_Collection {
...
 		 */
-		public function __construct( $slug, $data_or_file ) {
+		public function __construct( string $slug, array $args ) {
 			$this->slug = sanitize_title( $slug );
 			if ( $this->slug !== $slug ) {
 				_doing_it_wrong(
 					__METHOD__,
 					/* translators: %s: Font collection slug. */
-					sprintf( __( 'Font collection slug "%s" is not valid. Slugs must use only alphanumeric characters, dashes, and underscores.', 'gutenberg' ), $slug ),
+					sprintf( __( 'Font collection slug "%s" is not valid. Slugs must use only alphanumeric characters, dashes, and underscores.' ), $slug ),
```
Comment (vcanales) [inline lib/compat/wordpress-6.5/fonts/class-wp-font-collection.php]: Wondering if we need to add `'gutenberg'` back in this case?

Comment (matiasbenedetto) [reply lib/compat/wordpress-6.5/fonts/class-wp-font-collection.php]: yes, thanks for pointing that out

Comment (matiasbenedetto) [reply lib/compat/wordpress-6.5/fonts/class-wp-font-collection.php]: added that back here: https://github.com/WordPress/gutenberg/pull/59256/commits/fa6a219210bbb91bf59d9b8f225088855ce37270

File: lib/compat/wordpress-6.5/fonts/fonts.php [hunk 2]
```diff
@@ -157,7 +159,36 @@ function gutenberg_register_font_collections() {
 	if ( null !== WP_Font_Library::get_instance()->get_font_collection( 'google-fonts' ) ) {
 		return;
 	}
-	wp_register_font_collection( 'google-fonts', 'https://s.w.org/images/fonts/17.7/collections/google-fonts-with-preview.json' );
+	wp_register_font_collection(
+		'google-fonts',
+		array(
+			'name'          => _x( 'Google Fonts', 'font collection name' ),
```
Comment (creativecoder) [inline lib/compat/wordpress-6.5/fonts/fonts.php]: All of these strings need the 'gutenberg' text domain as well.

**Review by creativecoder — CHANGES_REQUESTED** [review]

**Comment by pbking** [comment]
image

If I pull out the category from the collection it's missing from the drop down. (Above `Sans Serif` was removed.)  I suspect that's as-expected, and that's similar behavior to before this change.  But with the category specifics now being provided separately from the font face assignment to those collections I just wanted to make sure.

If the above is as expected then this tests well for me.

**Comment by matiasbenedetto** [comment]
> If I pull out the category from the collection it's missing from the drop down.

@pbking yes, that's expected.

File: lib/compat/wordpress-6.5/fonts/fonts.php (hunk 2 above)
Comment (matiasbenedetto) [reply lib/compat/wordpress-6.5/fonts/fonts.php]: added here: https://github.com/WordPress/gutenberg/pull/59256/commits/450f9f5302df7f25cfd4c88c3abc64fb9ce31180

**Review by creativecoder — APPROVED** [review]
Thanks for adding the domain to the new text strings.

**Comment by getdave** [comment]
I rebased this and now hopefully all the tests will be ✅ and auto merge will handle.

**Pull request merged**